app/connectors/CalculatorConnector.scala

```diff
@@ -20,84 +20,91 @@ import constructors.resident.{properties => propertyConstructor}
 import models.resident._
 import models.resident.properties._
 import uk.gov.hmrc.http.HttpReads.Implicits._
-import uk.gov.hmrc.http.{HeaderCarrier, HttpClient}
+import uk.gov.hmrc.http.client.HttpClientV2
+import uk.gov.hmrc.http.{HeaderCarrier, StringContextOps}
 import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
-
 import java.time.LocalDate
 import javax.inject.{Inject, Singleton}
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
 class CalculatorConnector @Inject()(val servicesConfig: ServicesConfig,
-                          val http: HttpClient)(implicit val ec: ExecutionContext) {
-
-  val serviceUrl: String = servicesConfig.baseUrl("capital-gains-calculator")
+                                    val http: HttpClientV2)(implicit val ec: ExecutionContext) {
+  val headers: Seq[(String, String)] = Seq("Accept" -> "application/vnd.hmrc.1.0+json")
 
   implicit val hc: HeaderCarrier = HeaderCarrier()
+  private val serviceUrl: String = servicesConfig.baseUrl("capital-gains-calculator")
 
-  val headers = Seq("Accept" -> "application/vnd.hmrc.1.0+json")
-
-  def getMinimumDate()(implicit hc : HeaderCarrier): Future[LocalDate] = {
-    http.GET[LocalDate](s"$serviceUrl/capital-gains-calculator/minimum-date", headers = headers)
+  def getMinimumDate()(implicit hc: HeaderCarrier): Future[LocalDate] = {
+    http.get(url"$serviceUrl/capital-gains-calculator/minimum-date").setHeader("Accept" -> "application/vnd.hmrc.1.0+json")
```

Comment:
we should use `.transform(_.addHttpHeaders(setHeaders(): _*))` instead of `setHeaders()` .  Please refer to the slack comm https://hmrcdigital.slack.com/archives/C070VQGNKST/p1728310249501989?thread_ts=1728294830.628859&cid=C070VQGNKST